model/src/main/proto/feedback_reporting.proto

```diff
@@ -0,0 +1,136 @@
+syntax = "proto3";
+
+package model;
+
+import "profile.proto";
+
+option java_package = "org.oppia.android.app.model";
+option java_multiple_files = true;
+
+// Top-level proto to store feedback reports locally when there is no network connection to upload
+// them to the remote service
+message FeedbackReportingDatabase {
+  repeated FeedbackReport reports_list = 1;
+}
+
+// Corresponds to an entire feedback report with user reported information and collected app,
+// device, and build information.
+message FeedbackReport {
+  // The date the report was created.
+  int64 report_date_ms = 1;
+  FeedbackReportingForm feedback_form = 2;
+  FeedbackReportingDeviceInfo device_info = 3;
+  FeedbackReportingBuildInfo build_info = 4;
+  FeedbackReportingAppInfo app_info = 5;
+}
+
+// Information supplied by the user in the feedback report.
+message FeedbackReportingForm {
+  oneof report_type {
+    Suggestion suggestion = 1;
+    Issue issue = 2;
+    Crash crash = 3;
+  }
+}
+
+// A suggestion feedback report.
+message Suggestion {
+  enum SuggestionCategory {
+    CATEGORY_UNSPECIFIED = 0;
+    FEATURE_SUGGESTION = 1;
+    LANGUAGE_SUGGESTION = 2;
+    OTHER_SUGGESTION = 3;
+  }
+  SuggestionCategory suggestion_category = 1;
+  // The text input provided by the user.
+  string user_submitted_suggestion = 2;
+}
+
+// An issue feedback report.
+message Issue {
+  enum IssueCategory {
+    CATEGORY_UNSPECIFIED = 0;
+    LESSONS_QUESTIONS_ISSUE = 1;
+    LANGUAGE_ISSUE = 2;
+    LANGUAGE_AUDIO_ISSUE = 3;
+    LANGUAGE_TEXT_ISSUE = 4;
+    TOPICS_ISSUE = 5;
+    PROFILE_ISSUE = 6;
+    OTHER_ISSUE = 7;
+  }
+  IssueCategory issue_category = 1;
+  // List of checkbox items selected by user in the feedback report.
+  repeated string issues_list = 2;
+  // The user's text input if they select "other" for any option
+  string other_user_input = 3;
+}
+
+// A crash report.
+message Crash {
+  enum CrashLocation {
+    LOCATION_UNSPECIFIED = 0;
+    LESSON_PLAYER = 1;
+    PRACTICE_QUESTIONS = 2;
+    TOPICS_LIST = 3;
+    OPTIONS_ACTIVITY = 4;
+    PROFILE_ACTIVITY = 5;
+    OTHER = 6;
+  }
+  CrashLocation crash_location = 1;
+  // The user's text input for explaining what happened.
+  string crash_explanation = 2;
+}
+
+// Information collected about the user's device.
+message FeedbackReportingDeviceInfo {
+  string package_name = 1;
+  string package_code = 2;
+  string country_locale = 3;
+  string language_locale = 4;
+}
+
+// Information collected about the device's build.
+message FeedbackReportingBuildInfo {
+  enum NetworkType {
+    NETWORK_TYPE_UNSPECIFIED = 0;
+    WIFI = 1;
+    CELLULAR = 2;
+  }
+  string device_model = 1;
+  string sdk_version = 2;
+  string device_brand = 3;
+  string build_fingerprint = 4;
+  string phone_type = 5;
+  NetworkType network_type = 6;
+  string device_language = 7;
```

Comment:
How do this, language, & country locales differ?

Reply:
Country locale, language locale, and device language can all differ, e.g. French-Canada might have language locale = French, country locale = Canada, and device language = something completely different. The combination of these could help us understand the user better and then better contextualize the report. WDYT?

Reply:
That makes sense, but similar to my question above: can you point to the specific Android APIs used for each so that I can better understand how we're retrieving these values?

Reply:
[Locale](https://developer.android.com/reference/java/util/Locale.html) can give the [country](https://developer.android.com/reference/java/util/Locale.html#getISOLanguages()) and [language](https://developer.android.com/reference/java/util/Locale.html#getISO3Language()) codes as Strings (not all 3166 codes though, will make that change in the comments). It might be feasible to use an enum for the 3166 codes but that's a lot of codes to enumerate and since the API provides the string it seems reasonable to just pull directly form that return value.

Reply:
Ah I see your below comment regarding this. It looks the locale only gives information about the device's locale not the location locale so this device_language would be the same as language_locale. Will omit one of them.

Reply:
Removed device_language since it's the same as language_locale, keeping country_locale as well though it doesn't appear that this is the device's physical location (this requires location services that we probably don't want to be asking users for, not sure if it'll help right now).

Reply:
Yup, planning to use ISO3 codes for both language and country. Based on [this SO post](https://stackoverflow.com/questions/14389349/android-get-current-locale-not-default) the locale can be set by the user but does represent the country and language that they choose to use on the device, although not location.

Updated comment to hopefully make this clearer.

Reply:
Thanks for clarifying in the comment. Left one follow-up to further clarify the context of the comments.